pkg/apis/pgv2.percona.com/v2/perconapgcluster_types.go

```diff
@@ -393,6 +393,82 @@ type PerconaPGClusterStatus struct {
 	Host string `json:"host"`
 }
 
+type Backups struct {
+	// pgBackRest archive configuration
+	// +kubebuilder:validation:Required
+	PGBackRest PGBackRestArchive `json:"pgbackrest"`
+}
+
+func (b Backups) ToCrunchy() crunchyv1beta1.Backups {
+	return crunchyv1beta1.Backups{
+		PGBackRest: crunchyv1beta1.PGBackRestArchive{
+			Metadata:      b.PGBackRest.Metadata,
+			Configuration: b.PGBackRest.Configuration,
+			Global:        b.PGBackRest.Global,
+			Image:         b.PGBackRest.Image,
+			Jobs:          b.PGBackRest.Jobs,
+			Repos:         b.PGBackRest.Repos,
+			RepoHost:      b.PGBackRest.RepoHost,
+			Manual:        b.PGBackRest.Manual,
+			Restore:       b.PGBackRest.Restore,
+			Sidecars:      b.PGBackRest.Containers,
+		},
+	}
+}
+
+type PGBackRestArchive struct {
+	// +optional
+	Metadata *crunchyv1beta1.Metadata `json:"metadata,omitempty"`
+
+	// Projected volumes containing custom pgBackRest configuration.  These files are mounted
+	// under "/etc/pgbackrest/conf.d" alongside any pgBackRest configuration generated by the
+	// PostgreSQL Operator:
+	// https://pgbackrest.org/configuration.html
+	// +optional
+	Configuration []corev1.VolumeProjection `json:"configuration,omitempty"`
+
+	// Global pgBackRest configuration settings.  These settings are included in the "global"
+	// section of the pgBackRest configuration generated by the PostgreSQL Operator, and then
+	// mounted under "/etc/pgbackrest/conf.d":
+	// https://pgbackrest.org/configuration.html
+	// +optional
+	Global map[string]string `json:"global,omitempty"`
+
+	// The image name to use for pgBackRest containers.  Utilized to run
+	// pgBackRest repository hosts and backups. The image may also be set using
+	// the RELATED_IMAGE_PGBACKREST environment variable
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// Jobs field allows configuration for all backup jobs
+	// +optional
+	Jobs *crunchyv1beta1.BackupJobs `json:"jobs,omitempty"`
+
+	// Defines a pgBackRest repository
+	// +kubebuilder:validation:MinItems=1
+	// +listType=map
+	// +listMapKey=name
+	Repos []crunchyv1beta1.PGBackRestRepo `json:"repos"`
+
+	// Defines configuration for a pgBackRest dedicated repository host.  This section is only
+	// applicable if at least one "volume" (i.e. PVC-based) repository is defined in the "repos"
+	// section, therefore enabling a dedicated repository host Deployment.
+	// +optional
+	RepoHost *crunchyv1beta1.PGBackRestRepoHost `json:"repoHost,omitempty"`
+
+	// Defines details for manual pgBackRest backup Jobs
+	// +optional
+	Manual *crunchyv1beta1.PGBackRestManualBackup `json:"manual,omitempty"`
+
+	// Defines details for performing an in-place restore using pgBackRest
+	// +optional
+	Restore *crunchyv1beta1.PGBackRestRestore `json:"restore,omitempty"`
+
+	// Configuration for pgBackRest sidecar containers
+	// +optional
+	Containers *crunchyv1beta1.PGBackRestSidecars `json:"containers,omitempty"`
```

Comment:
This change was made in order so we have a aligned API, meaning we want here to have `containers` rather than `sidecars` that come with using Crunchy backup type. Everything else stays the same.

The question is, should we maybe first deprecate `Sidecars` that was technically available to users, but in our reference docs we don't list this field, so I'm pretty sure we are OK just to remove it.

@egegunes @hors wdyt?

Reply:
We do not have it but people can use it :) I vote for deprecation.

Reply:
You think there could be a user that uses it?

Reply:
@hors in our reference docs we didn't specify that we have this option, this means that the user had to know what we internally use for backups and plus that he knows crunchy spec. 

I'm pretty sure that no one used this field and I would like to skip all the deprecations ceremony only because I'm pretty sure we don't need it. And actually since we didn't specify this option nowhere, what should we deprecate in the first place :)

Reply:
I was in no-need-deprecate camp but now thinking about and we were exposing this field through our CR. I think we need to deprecate.

Reply:
@egegunes, we are not exposing it in our CR and we don't specify it in our reference docs.

Reply:
[ac21dc0](https://github.com/percona/percona-postgresql-operator/pull/761/commits/ac21dc01546ce4e8af1361ecc04a08f77f7509e0)